homeassistant/components/melnor/config_flow.py

```diff
@@ -47,10 +56,59 @@ async def async_step_bluetooth(
         address = discovery_info.address
 
         await self.async_set_unique_id(address)
-
         self._abort_if_unique_id_configured(updates={CONF_ADDRESS: address})
 
         self._discovered_address = address
 
         self.context["title_placeholders"] = {"name": address}
         return await self.async_step_bluetooth_confirm()
+
+    async def async_step_pick_device(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the step to pick discovered device."""
+
+        if user_input is not None:
+            return self._create_entry(user_input[CONF_ADDRESS])
+
+        current_addresses = self._async_current_ids()
+        for connectable in (True, False):
+            for discovery_info in async_discovered_service_info(self.hass, connectable):
+                if discovery_info.manufacturer_id == MANUFACTURER_ID and any(
+                    manufacturer_data.startswith(MANUFACTURER_DATA_START)
+                    for manufacturer_data in discovery_info.manufacturer_data.values()
+                ):
+
+                    address = discovery_info.address
+                    if (
+                        address in current_addresses
+                        or address in self._discovered_addresses
+                    ):
+                        continue
+
+                    self._discovered_addresses.append(address)
+
+        addresses = {
+            address
+            for address in self._discovered_addresses
+            if address not in current_addresses
+        }
+
+        # Check if there is at least one device
+        if not addresses:
+            return self.async_abort(reason="no_devices_found")
+
+        return self.async_show_form(
+            step_id="pick_device",
+            data_schema=vol.Schema({vol.Required(CONF_ADDRESS): vol.In(addresses)}),
+        )
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+
+        if self._discovered_addresses is None:
```

Comment:
You'll need to initialize `self._discovered_addresses` when the flow starts otherwise its always going to abort

Reply:
Hmm yeah, this isn't really a good spot for this check. Makes more sense to handle it in the device picker builder where I have all the state already.